CefSharp.WinForms.Example/BrowserTabUserControl.cs

```diff
@@ -232,71 +231,90 @@ private void OnIsBrowserInitializedChanged(object sender, IsBrowserInitializedCh
                 // which running in a different thread we have to use a NativeWindow
                 if (multiThreadedMessageLoopEnabled)
                 {
-                    Task.Run(() =>
+                    SetupMessageInterceptor();
+                }
+            }
+        }
+
+        private void SetupMessageInterceptor()
+        {
+            if (messageInterceptor != null)
+            {
+                messageInterceptor.ReleaseHandle();
+                messageInterceptor = null;
+            }
+
+            Task.Run(async () =>
+            {
+                try
+                {
+                    while (true)
                     {
-                        try
+                        IntPtr chromeWidgetHostHandle;
+                        if (ChromeWidgetHandleFinder.TryFindHandle(browserHandle, out chromeWidgetHostHandle))
                         {
-                            while (true)
+                            messageInterceptor = new ChromeWidgetMessageInterceptor((Control)Browser, chromeWidgetHostHandle, message =>
                             {
-                                IntPtr chromeWidgetHostHandle;
-                                if (ChromeWidgetHandleFinder.TryFindHandle(browserHandle, out chromeWidgetHostHandle))
+                                const int WM_MOUSEACTIVATE = 0x0021;
+                                const int WM_NCLBUTTONDOWN = 0x00A1;
+                                const int WM_LBUTTONDOWN = 0x0201;
+                                const int WM_DESTROY = 0x0002;
+
+                                // Render process switch happened, need to find the new handle
```

Comment:
Anything more than this one?
You mean a `///<summary>` or?

Reply:
I think a summary of what's going on would be helpful. I can add that later if you'd like

Reply:
Yeah, a followup commit would probably be the best